doc/qos/ECN_and_WRED_statistics_HLD.md

```diff
@@ -0,0 +1,342 @@
+    - [High-Level Design](#high-level-design)
+      - [Changes in COUNTERS_DB](#changes-in-counters_db)
+      - [Changes in FLEX_COUNTER_DB](#changes-in-flex_counter_db)
+      - [Changes in STATE_DB](#changes-in-state_db)
+      - [Changes in Orchagent](#changes-in-orchagent)
+      - [CLI Changes](#cli-changes)
+        - [show queue counters on a ECN statistics supported platform](#show-queue-counters-on-a-ecn-statistics-supported-platform)
+        - [show queue counters on a platform which does not support ECN statistics](#show-queue-counters-on-a-platform-which-does-not-support-ecn-statistics)
+        - [show interface counters on a WRED drop statistics supported platform](#show-interface-counters-on-a-wred-drop-statistics-supported-platform)
+        - [show interface counters on a platform which does not support WRED drop statistics](#show-interface-counters-on-a-platform-which-does-not-support-wred-drop-statistics)
+    - [SAI API](#sai-api)
+    - [Configuration and management](#configuration-and-management)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+
+### Revision
+
+| Rev | Date     | Author                      | Change Description |
+|:---:|:--------:|:---------------------------:|--------------------|
+| 0.1 |23/Feb/23 | Rajesh Perumal **(Marvell)**| Initial Version    |
+
+### Scope
+
+This document provides the high level design for ECN Queue Statistics, WRED Port Statistics in SONiC
+
+### Abbreviations
+
+
+
+|  Abbreviation | Description                     |
+|:-------------:|---------------------------------|
+| __ECN__       | Explicit Congestion Notification|
+| __WRED__      | Weighted Random Early Detection |
+| __CLI__       | Command Line interface          |
+
+
+### Overview
+
+The main goal of this feature is to provide better WRED impact visibility in SONiC by providing a mechanism to count the packets that are discarded or ECN-marked due to WRED.
+
+The other goal of this feature is to display these statistics only if the underlying platform supports it. [CLI changes](#cli-changes) section explains this in detail. Every platform may have unique statistics capabilities, and they change over time, and so it is important for this feature to be capability based.
+
+We will accomplish both the goals by adding support for per-queue ECN marked packets/bytes counters and per-port WRED dropped packets. Existing “queue counters” and “show interface counters detailed” CLI  will be enhanced for displaying these statistics.
+
+
+### Requirements
+
+1. Support per-queue total ECN marked packets counters
+2. Support per-queue total ECN marked byte counters
+3. Support per-port WRED dropped packets counters (per-color and total count)

```

Comment:
Hi, Is there plan for per-queue WRED dropped counters?

Reply:
I have added support for queue level WRED dropped stats in the updated HLD. For the case of port level ECN marked stats, there is not a requirement from customers for this stats. However we have planned to take it up in the next phase of enhancement.
Hope this is fine with you.

Reply:
Hi @rpmarvell 
can you add the port level ECN marked counter (SAI_PORT_STAT_ECN_MARKED_PACKETS) to the HLD and mention it will be supported in the next phase of enhancement? I think it is important for platforms that do not support queue-level ECN marked packets.
Also, can you share the plan for the next phase of enhancement?
thanks

Reply:
I have updated the HLD with Port ECN marked stats. Also mentioned that it will be taken up in next phase of enhancement.
I will discuss internally and provide the plan once it is finalized. Thank you!